server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileExporters.java

```diff
@@ -118,14 +121,19 @@ public String export(String profileKey, String tool) {
   }
 
   private RulesProfile wrap(QualityProfileDto profile) {
+    DbSession dbSession = dbClient.openSession(false);
     RulesProfile target = new RulesProfile(profile.getName(), profile.getLanguage());
-    for (Iterator<ActiveRule> activeRuleIterator = loader.findActiveRulesByProfile(profile.getKey()); activeRuleIterator.hasNext();) {
-      ActiveRule activeRule = activeRuleIterator.next();
-      Rule rule = ruleFinder.findByKey(activeRule.key().ruleKey());
-      org.sonar.api.rules.ActiveRule wrappedActiveRule = target.activateRule(rule, RulePriority.valueOf(activeRule.severity()));
-      for (Map.Entry<String, String> entry : activeRule.params().entrySet()) {
-        wrappedActiveRule.setParameter(entry.getKey(), entry.getValue());
+    try {
+      for (ActiveRuleDto activeRule : dbClient.activeRuleDao().selectByProfileKey(dbSession, profile.getKey())) {
+        Rule rule = ruleFinder.findByKey(activeRule.getKey().ruleKey());
+        org.sonar.api.rules.ActiveRule wrappedActiveRule = target.activateRule(rule, RulePriority.valueOf(activeRule.getSeverityString()));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "RulePriority"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)